contents/tutorials/nextjs-surveys.md

```diff
@@ -0,0 +1,536 @@
+    setShowSurvey(false);
+    console.log("Survey dismissed!");
+    localStorage.setItem(`hasInteractedWithSurvey_${surveyID}`, 'true');
+  };
+
+  const handleSubmit = (value) => {
+    setShowSurvey(false);
+    console.log("User submitted:", value);
+    localStorage.setItem(`hasInteractedWithSurvey_${surveyID}`, 'true');  
+  };
+
+  // ... rest of your code ...
+```
+
+#### 3. Capture interactions from it.
+
+The final step in setting up our survey is capturing interactions. This enables us to analyze the results in PostHog. 
+
+There are 3 events to capture:
+
+1. `"survey shown"`
+2. `"survey dismissed"`
+3. `"survey sent"` (for responses)
+
+You can capture these events using `posthog.capture()`:
+
+```react
+  // ... rest of your code ...
+
+  const handleDismiss = () => {
+    setShowSurvey(false);
+    console.log("Survey dismissed!");
+    localStorage.setItem(`hasInteractedWithSurvey_${surveyID}`, 'true');
+    posthog.capture("survey dismissed", {
+      $survey_id: surveyID // required
+    })
+  };
+
+  const handleSubmit = (value) => {
+    setShowSurvey(false);
+    console.log("User submitted:", value);
+    localStorage.setItem(`hasInteractedWithSurvey_${surveyID}`, 'true');  
+    posthog.capture("survey sent", {
+      $survey_id: surveyID, // required
+      $survey_response: value // required
+    })
+  };
+
+  useEffect(() => {
+    if (posthog && surveyID && showSurvey) {
+      posthog.capture("survey seen", {
+        $survey_id: surveyID // required
+      })
+    }
+  }, [showSurvey, surveyID, posthog])
+
+ // ... rest of your code ...
+```
+
+Altogether, your code should look like this:
+
+```react
+// app/page.js
+'use client'
+
+import { useState, useEffect } from 'react'
+import styles from './page.module.css'
+import Survey from './Survey';
+import { usePostHog } from 'posthog-js/react';
+
+export default function Home() {
+  const [showSurvey, setShowSurvey] = useState(true);
+  const [surveyTitle, setSurveyTitle] = useState(false);
+  const [surveyID, setSurveyID] = useState(false);
+  const posthog = usePostHog()
+
+  useEffect(() => {
+    posthog.getActiveMatchingSurveys((surveys) => {
+      if (surveys.length > 0) {
+        const survey = surveys[0];
+        setSurveyID(survey.id);
+        setSurveyTitle(survey.questions[0].question)
+      }
+    }); 
+  }, [posthog])
+
+  const handleDismiss = () => {
+    setShowSurvey(false);
+    console.log("Survey dismissed!");
+    localStorage.setItem(`hasInteractedWithSurvey_${surveyID}`, 'true');
+    posthog.capture("survey dismissed", {
+      $survey_id: surveyID // required
+    })
+  };
+
+  const handleSubmit = (value) => {
+    setShowSurvey(false);
+    console.log("User submitted:", value);
+    localStorage.setItem(`hasInteractedWithSurvey_${surveyID}`, 'true');  
+    posthog.capture("survey sent", {
+      $survey_id: surveyID, // required
+      $survey_response: value // required
+    })
+  };
+
+  useEffect(() => {
```

Comment:
This seems to run before the `hasInteractedWithSurvey` check (at least in my implementation), meaning there are "seen survey" events after they submit and without giving them a chance to submit. Need to make sure this works.

Reply:
Fixed. Need to change default showSurvey state from true to false i.e.

`const [showSurvey, setShowSurvey] = useState(true);` to `  const [showSurvey, setShowSurvey] = useState(false);